api/mapping/models.py

```diff
@@ -354,3 +366,43 @@ class NLPModel(models.Model):
 
     def __str__(self):
         return str(self.id)
+
+
+class Dataset(BaseModel):
+    """
+    Model for datasets which contain scan reports.
+    """
+
+    id = models.UUIDField(primary_key=True, editable=False, default=uuid.uuid4)
+    name = models.CharField(max_length=100)
+    # `projects` field added by M2M field in `Project`
+    # `scan_reports` field added by FK field in `ScanReport`
+
+    class Meta:
+        verbose_name = "Dataset"
+        verbose_name_plural = "Datasets"
+
+    def __str__(self) -> str:
+        return self.name
```

Comment:
`self.name` is more readable than `self.id` especially when you use UUIDs.

Reply:
I'd be interested to know why this was decided (before my time). Calum, can you remember the argument (or who was in it)?

Reply:
@vpnu might remember, it might have been less of an argument and more we were just told to change them all to `str(self.id)` without much explanation

Reply:
Yeah it was just decided, don't remember being told why either. Before that we used to have self.name instead